controllers/app_controller.go

```diff
@@ -63,6 +67,18 @@ func (r *AppReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	// Check for HelmRelease with same name (migration warning)
+	// This is informational only - App controller continues to work for backwards compatibility
+	if hasHelmRelease, err := r.hasMatchingHelmRelease(ctx, req.NamespacedName); err != nil {
+		log.Error(err, "Failed to check for matching HelmRelease")
+	} else if hasHelmRelease {
+		klog.Warningf("a HelmRelease with same name as this App CR exists (namespace=%s, name=%s). The HelmRelease will be given preference, and this App CR will be ignored. Refer to <put a doc link here> for more information.",
```

Comment:
Please revert the extra logging library https://github.com/giantswarm/dex-operator/pull/141#discussion_r2590309611

Reply:
And also link to a doc there :smile: